wandb/sdk/wandb_run.py

```diff
@@ -1194,13 +1204,23 @@ def _summary_update_callback(self, summary_record: SummaryRecord) -> None:
         if self._backend and self._backend.interface:
             self._backend.interface.publish_summary(summary_record)
 
+    def _on_progress_get_summary(self, handle: MailboxProgress) -> None:
+        line = "Waiting for run.summary data...\r"
+        # TODO: use printer?
```

Comment:
would it be hard to use printer here? (I know we talked that, so up to you)

Reply:
i changed it to use printer and added some ratelimiting, then i decided to defer to a later mailbox phase.
I am worried about being too spammy outputing "waiting for summary" messages.

rough plan is to only show this progress indication after N seconds of waiting for this summary event.